public/js/actions/notifications.js

```diff
@@ -1,14 +1,58 @@
 import * as actionTypes from 'constants/action-types';
+import * as errorCodes from 'constants/error-codes';
 
+import { gettext } from 'utils';
 
-export function addNotification(text, { type='info', userDismissable=false,
-                                autoHide=true } = {}) {
+
+export function addNotification({ text, errorCode,
+                                  type='info', userDismissable,
+                                  autoHide } = {}) {
+
+  if (type === 'error' && !text) {
+    text = gettext('Internal error. Please try again later.');
+  }
+
+  if (type === 'info' && !text) {
+    throw new Error('An info notification requires text');
+  }
+
+  // Default to error type being userDismissable.
+  if (type === 'error' && typeof userDismissable === 'undefined') {
+    userDismissable = true;
+  }
+
+  // Default info type to being hidden on delay.
+  if (type === 'info' && typeof autoHide === 'undefined') {
+    autoHide = true;
+  }
```

Comment:
duh, I see you have separate functions already. Maybe move all the `if (type==='error')` checks and so on to each respective function? You could just make `addNotification` private so that it's never used directly.